aws_auth.tf

```diff
@@ -41,9 +41,13 @@ data "template_file" "worker_role_arns" {
     )
   }
 }
+locals {
+  kubeconfig_filename = concat(local_file.kubeconfig.*.filename, [""])[0]
+}
 
 resource "kubernetes_config_map" "aws_auth" {
-  count = var.create_eks && var.manage_aws_auth ? 1 : 0
+  depends_on = [aws_eks_cluster.this]
```

Comment:
Is this explicit `depends_on` required? The provider doesn't work before the cluster's creation.

Reply:
It depends on how the kubernetes provider is set up(?) I wasn't sure it wasn't impossible make dependence on cluster opaque to terraform, depending on how the provider is setup, so thought explicit dependency would be useful to avoid bug in edge case.

Reply:
Very true. It's not necessarily implicit from other variables.